docs/docs/developer/frontend/best-practices.mdx

```diff
@@ -147,5 +147,267 @@ We do not recommend `React.memo()` usage because it does not solve the cause of
+
+#### Units of measurement
+
+Avoid using `px` or `rem` values directly within the styled components. The necessary values are generally already defined in the theme, so it’s recommended to make use of the theme for these purposes.
+
+#### Colors
+
+Refrain from introducing additional colors; instead, utilize the existing palette from the theme. Should there be a situation where the palette does not align, kindly leave a comment so that it can be rectified.
+
+
+```tsx
+// ❌ Bad, directly specifies style values without utilizing the theme
+const StyledButton = styled.button`
+  color: #333333;
+  font-size: 1rem;
+  font-weight: 400;
+  margin-left: 4px;
+  border-radius: 50px;
+`;
+```
+
+```tsx
+// ✅ Good, utilizes the theme
+const StyledButton = styled.button`
+  color: ${({ theme }) => theme.font.color.primary};
+  font-size: ${({ theme }) => theme.font.size.md};
+  font-weight: ${({ theme }) => theme.font.weight.regular};
+  margin-left: ${({ theme }) => theme.spacing(1)};
+  border-radius:  ${({ theme }) => theme.border.rounded};
+`;
+```
+
+## Static Typing
+
+### No `interface`'s
+
+The project prioritizes the usage of `type` definitions over `interface`'s.
+
+```tsx
+// ❌ Bad, utilizes an interface
+interface User {
+  name: string;
+  age: number;
+}
+```
+
+```tsx
+// ✅ Good, utilizes a type
+type User = {
+  name: string;
+  age: number;
+};
+```
+
+### No `FC`/`FunctionComponent`
+
+As per project convention, please refrain from using `React.FC` or `React.FunctionComponent`. Instead, opt for defining functions with the format `FunctionName`.
+
+Secondly, avoid utilizing arrow function syntax for defining components. Please adhere to the format of exporting components using the `export function FunctionName() {}` syntax.
+
+```tsx
+/* ❌ - Bad, defines the component type annotations with `FC`
+ *    - With `React.FC`, the component implicitly accepts a `children` prop
+ *      even if it's not defined in the prop type. This might not always be
+ *      desirable, especially if the component doesn't intend to render
+ *      children.
+ *    - The component is defined as a constant that holds an arrow function
+ */
+const EmailField: React.FC<{
+  value: string;
+}> = ({ value }) => <TextInput value={value} disabled fullWidth />;
+```
+
+```tsx
+/* ✅ - Good, a separate type (OwnProps) is explicitly defined for the 
+ *      component's props
+ *    - This method doesn't automatically include the children prop. If
+ *      you want to include it, you have to specify it in OwnProps.
+ *    - The component is defined using a traditional function declaration
+ */ 
+type OwnProps = {
+  value: string;
+};
+
+function EmailField({ value }: OwnProps) {
+  return <TextInput value={value} disabled fullWidth />;
+}
+```
+
+### Fixed constant sets
+
+For situations where a value anticipates a fixed set of constants, it’s advisable to define enums over union types.
+
+**Why Opt for Enums Over Union Types**
```

Comment:
cc @charlesBochet I think we are still reconsidering this, I think we can merge as it is but let's chat again after then create an eslint rule if needed